src/engine/attributes/set.js

```diff
@@ -1,33 +1,34 @@
 import "../../../src/engine/helper/setMatcher.js";
-import { convertToString } from "../../../src/utils/lang/index.js";
 
 export default function setApply(val, userVal, op){
 
-	let vals = convertToString(val)
-	const ruleSet = new Set();
-	vals.split(",").map((item)=>{
-		ruleSet.add(item.trim())
-	}) 
+	if (userVal === undefined || userVal === null) {
+		throw new Error("Invalid set attribute value. Set attribute must be type of set or array.")
+	}
 
-	let uValues = convertToString(userVal)
-	let userValuesSet = new Set()
-	uValues.split(",").map((item)=>{	
-		userValuesSet.add(item.trim())
+	let userValuesSet;
+	if (Array.isArray(userVal)) {
```

Comment:
We should allow strings in set as well because in Javascript if someone passes string in Set constructor it will return set of unique value.
For eg:
var setObj = new Set("fooooooood"); 
//  Set(3) [ "f", "o", "d" ]

Reply:
users can pass in either array or set, unless I'm missing something.

Reply:
We are allowing either array or set only not strings right now.. we should allow strings as well